kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/bluetooth/le/BluetoothLeAdapter.java

```diff
@@ -0,0 +1,241 @@
+
+    /**
+     * Search for a BLE device with the specified name. The method will perform a BLE discovery for at most timeout
+     * seconds or until the device is found. It will return a Future instance and the discovered device can be
+     * retrieved using the get() method. The get(long timeout, TimeUnit unit) is not supported and acts as the get()
+     * method.
+     * 
+     * @param timeout
+     *            timeout in seconds for device discovery
+     * @param name
+     *            system name of the BLE device
+     * @return Future
+     */
+    public Future<BluetoothLeDevice> findDeviceByName(long timeout, String name);
+
+    /**
+     * Search for a BLE device with the specified address. The method will perform a BLE discovery for at most timeout
+     * seconds or until the device is found. When the device is found or the timeout is reached the consumer is used to
+     * get the device.
+     * 
+     * @param timeout
+     *            timeout in seconds for device discovery
+     * @param address
+     *            MAC address of the BLE device
+     * @param consumer
+     *            the consumer used to get the device
+     */
+    public void findDeviceByAddress(long timeout, String address, Consumer<BluetoothLeDevice> consumer);
+
+    /**
+     * Search for a BLE device with the specified name. The method will perform a BLE discovery for at most timeout
+     * seconds or until the device is found. When the device is found or the timeout is reached the consumer is used to
+     * get the device.
+     * 
+     * @param timeout
+     *            timeout in seconds for device discovery
+     * @param name
+     *            system name of the BLE device
+     * @param consumer
+     *            the consumer used to get the device
+     */
+    public void findDeviceByName(long timeout, String name, Consumer<BluetoothLeDevice> consumer);
+
+    /**
+     * Search for BLE devices. The method will perform a BLE discovery for timeout seconds. It will return a Future
+     * instance and the discovered devices can be retrieved using the get() method. The get(long timeout, TimeUnit unit)
+     * is not supported and acts as the get() method.
+     * 
+     * @param timeout
+     *            timeout in seconds for device discovery
+     * @return Future
+     */
+    public Future<List<BluetoothLeDevice>> findDevices(long timeout);
+
+    /**
+     * Search for BLE devices. The method will perform a BLE discovery for timeout seconds. When the timeout is reached
+     * the consumer is used to get the devices.
+     * 
+     * @param timeout
+     *            timeout in seconds for device discovery
+     * @param consumer
+     *            the consumer used to get the device
+     */
+    public void findDevices(long timeout, Consumer<List<BluetoothLeDevice>> consumer);
+
+    /**
+     * Stops a BLE discovery.
+     * 
+     */
+    public void stopDiscovery() throws KuraException;
```

Comment:
Most of the methods simply call the tinyb methods and these don't throw any exception. I'll modify the exceptions thrown in the remaining cases.